components/core/src/os/users/admincheck.c

```diff
@@ -0,0 +1,90 @@
+#include <windows.h>
+#include <winnt.h>
+#include <ShlObj.h>
+
+#pragma comment (lib,"shell32.lib")
+
+TOKEN_ELEVATION_TYPE s_elevationType = TokenElevationTypeDefault;
+BOOL                 s_bIsAdmin = FALSE;
+
+// returns 0 if an admin
+// returns 1 if not
+int GetElevationState()
```

Comment:
Thanks @chefsalim - first of all, hopefully this is temporary and we can move to straight ffi against, the requisite apis, but I left this an int rather than a bool in case I need to do more discreet separation of user elevation types later (could drop in an enum or something).